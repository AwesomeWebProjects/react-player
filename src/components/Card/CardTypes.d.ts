import type { ElementType, ReactNode } from 'react'

export interface Card {
  className?: string
  element?: ElementType
  children: Element | ElementType | ReactNode
}

