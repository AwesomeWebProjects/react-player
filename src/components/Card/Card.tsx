import { memo } from 'react'
import classNames from 'classnames'
import type { Card } from './CardTypes'
import styles from './styles.module.css'

const Card = ({ className, children, element: Element = 'div', ...rest }: Card) => (
  <Element {...rest} className={classNames(styles['card'], className)}>
    {children}
  </Element>
)

export default memo(Card)
