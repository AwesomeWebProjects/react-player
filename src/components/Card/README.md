```js
<div style={{ display: 'grid', rowGap: '24px' }}>
  <Card type="1">
    Lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet
  </Card>
  <Card type="2">
    Lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet
  </Card>
  <Card type="3">
    Lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet
  </Card>
  <Card type="4">
    Lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet
  </Card>
  <Card type="5">
    Lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet
  </Card>
  <Card type="6">
    Lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet
  </Card>
</div>
```
