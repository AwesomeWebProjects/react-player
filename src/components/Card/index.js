import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import style from './style.module.css'

const Card = ({ className, children, element: Element, type, ...rest }) => (
  <Element {...rest} className={classNames(style.card, className, style[`shadow-${type}`])}>
    {children}
  </Element>
)

Card.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.node,
    PropTypes.object,
  ]).isRequired,
  element: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.node,
    PropTypes.object,
  ]),
  type: PropTypes.oneOf(['flat', '1', '2', '3', '4', '5', '6']),
}

Card.defaultProps = {
  className: null,
  element: 'div',
  type: '2',
}

export default Card
