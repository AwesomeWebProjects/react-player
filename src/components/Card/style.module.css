.card {
  display: flex;
  flex-direction: column;
  border-radius: var(--begrowth-border-radius-l);
  height: auto;
  padding: var(--begrowth-spacing-m);
  background: var(--begrowth-color-white);
}

.shadow-1 {
  box-shadow: var(--begrowth-shadow-1);
}

.shadow-2 {
  box-shadow: var(--begrowth-shadow-2);
}

.shadow-3 {
  box-shadow: var(--begrowth-shadow-3);
}

.shadow-4 {
  box-shadow: var(--begrowth-shadow-4);
}

.shadow-5 {
  box-shadow: var(--begrowth-shadow-5);
}

.shadow-6 {
  box-shadow: var(--begrowth-shadow-6);
}

.shadow-flat {
  box-shadow: none;
}
