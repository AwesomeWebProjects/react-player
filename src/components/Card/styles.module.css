.card {
  @apply p-4 bg-white h-auto flex flex-col rounded-xl shadow-lg;
}
