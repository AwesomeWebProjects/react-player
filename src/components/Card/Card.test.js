/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import Card from './index'

test('Card: simple render test', () => {
  const component = renderer.create(
    <Card>
      <p>
        middle skin live grabbed daughter wish page apart influence breathing which friendly getting
        laid difficulty cut specific curious indeed third coat right forty shorter
      </p>
      <p>
        middle skin live grabbed daughter wish page apart influence breathing which friendly getting
        laid difficulty cut specific curious indeed third coat right forty shorter
      </p>
    </Card>,
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
