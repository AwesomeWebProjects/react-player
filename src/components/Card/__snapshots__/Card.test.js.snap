// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card: simple render test 1`] = `
<div
  className="card"
>
  <p>
    middle skin live grabbed daughter wish page apart influence breathing which friendly getting laid difficulty cut specific curious indeed third coat right forty shorter
  </p>
  <p>
    middle skin live grabbed daughter wish page apart influence breathing which friendly getting laid difficulty cut specific curious indeed third coat right forty shorter
  </p>
</div>
`;
